ui/raidboss/browser_tts_engine.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+const TTSEngineType = {
+  SpeechSynthesis: 0,
+  GoogleTTS: 1,
+};
+
+class TTSItem {
+  play() {}
+}
+
+class SpeechTTSItem extends TTSItem {
+  constructor(text) {
+    super();
+    this.text = text;
+    this.item = new SpeechSynthesisUtterance(text);
+  }
+
+  play() {
+    window.speechSynthesis.speak(this.item);
+  }
+}
+
+class GoogleTTSItem extends TTSItem {
+  constructor(text) {
+    super();
+    this.text = text;
+    let iframe = document.createElement('iframe');
+    // remove sandbox so we can modify contents/call play on audio element later
+    iframe.removeAttribute('sandbox');
+    iframe.style.display = 'none';
+    document.body.appendChild(iframe);
+    let encText = encodeURIComponent(text);
+    iframe.contentDocument.body.innerHTML = '<audio src="https://translate.google.com/translate_tts?ie=UTF-8&client=tw-ob&tl=en&q=' + encText + '" id="TTS">';
+    this.item = iframe.contentDocument.body.firstElementChild;
+  }
+
+  play() {
+    this.item.play();
+  }
+}
+
+class BrowserTTSEngine {
+  constructor() {
+    engineType = null;
```

Comment:
`this.engineType` and `this.ttsItems`

Reply:
This is fixed in my local git now, though it worked fine without the `this.` so it was more of a linting thing?

Reply:
This was an error for me.  I don't know how that worked with 'use strict'.